File: Mbs.Api/Models/Trading/Instruments/Inav.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using Mbs.Trading.Currencies;

namespace Mbs.Api.Models.Trading.Instruments
{
    /// <summary>
    /// An additional information for Indicative Net Asset Values.
    /// </summary>
    [DataContract]
    public class Inav : IValidatableObject
    {
        /// <summary>
        /// Gets or sets a currency code.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public CurrencyCode Currency { get; set; } = CurrencyCode.Xxx;

        /// <summary>
        /// Gets or sets a target instrument reference.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public InstrumentReference Target { get; set; }

        /// <inheritdoc />
        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }
}

File: Mbs.Api.Host.Ng/ClientApp/src/app/shared/mbs/instruments/types/inav.ts
import { CurrencyCode } from '../../currencies/currency-code.enum';
import { InstrumentReference } from './instrument-reference';

/** An additional information for Indicative Net Asset Values. */
export class Inav {
  /** A currency code. */
  currency!: CurrencyCode;

  /** A target instrument reference. */
  target?: InstrumentReference;

  constructor(data?: Inav) {
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property)) {
          (this as any)[property] = (data as any)[property];
        }
      }
      this.target = (data.target && !(data.target).toJSON) ? new InstrumentReference(data.target) :
        (this.target as InstrumentReference);
    }
  }

  toJSON(data?: any) {
    data = typeof data === 'object' ? data : {};
    /* eslint-disable @typescript-eslint/dot-notation */
    data['Currency'] = this.currency;
    data['Target'] = this.target ? this.target.toJSON() : (undefined as any);
    /* eslint-enable @typescript-eslint/dot-notation */
    return data;
  }
}
